tests/PHPUnit/Integration/CronArchive/QueueConsumerTest.php

```diff
@@ -715,6 +715,137 @@ public function test_skipSegmentsToday()
         $this->assertEquals($uniqueInvalidationDescs, $invalidationDescs, "Found duplicate archives being processed.");
     }
 
+    public function testDeletedSegment()
+    {
```

Comment:
If I see that correctly this test checks how a segment is handled that was removed from the segment table.
This actually doesn't reflect what the fix is meant for. It should actually ensure that segments that might still exist, but got invalid (e.g. due to removing a plugin) don't break anything.
So a proper test would be to (maybe manually) insert invalidations for an invalid segment (e.g. a condition that is invalid like `InvalidSegment==val`) and check if running the cron doesn't fail, but silently removes such invalidations.

Reply:
@sgiehl just checking, if I am right. So I have to manually insert a record into `archive_invalidations` table with invalid segment, but it seems like the `archive_invalidations` reset all the time the manually insert will be clear.

Reply:
The archiving and invalidations stuff is quite complex. It's even hard for me to say something without looking deeper into it. But I would guess that it might work with adding it to the ReArchiveList like here:
https://github.com/matomo-org/matomo/blob/fe255ed247035b2deaadd058cbbaf041a92d3ad7/core/Archive/ArchiveInvalidator.php#L541-L548